gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/metrics/kafka/KafkaAvroSchemaRegistry.java

```diff
@@ -85,7 +85,22 @@ public KafkaAvroSchemaRegistry(Properties props) {
     GenericObjectPoolConfig config = new GenericObjectPoolConfig();
     config.setMaxTotal(objPoolSize);
     config.setMaxIdle(objPoolSize);
-    this.httpClientPool = new GenericObjectPool<>(new HttpClientFactory(), config);
+
+    HttpClientFactory factory = new HttpClientFactory();
+
+    if (this.props.containsKey(ConfigurationKeys.KAFKA_SCHEMA_REGISTRY_HTTPCLIENT_SO_TIMEOUT)) {
+      String soTimeout = this.props.getProperty(ConfigurationKeys.KAFKA_SCHEMA_REGISTRY_HTTPCLIENT_SO_TIMEOUT);
+      Preconditions.checkArgument(Integer.parseInt(soTimeout) >= 0);
```

Comment:
We should allow negative values. This is useful when overriding a common value in Azkaban.